src/transformers/models/cvt/modeling_tf_cvt.py

```diff
@@ -0,0 +1,945 @@
+
+    def convolution(self, hidden_state: tf.Tensor) -> tf.Tensor:
+        # Custom padding to match the model implementation in PyTorch
+        height_pad = width_pad = (self.pad_value, self.pad_value)
+        hidden_state = tf.pad(hidden_state, [(0, 0), height_pad, width_pad, (0, 0)])
+        hidden_state = self.conv(hidden_state)
+        return hidden_state
+
+    def call(self, hidden_state: tf.Tensor, training: bool = False) -> tf.Tensor:
+        hidden_state = self.convolution(hidden_state)
+        hidden_state = self.Normalization(hidden_state, training=training)
+        return hidden_state
+
+
+class TFCvtSelfAttentionLinearProjection(tf.keras.layers.Layer):
+    """Linear projection layer used to flatten tokens into 1D."""
+
+    def call(self, hidden_state: tf.Tensor) -> tf.Tensor:
+        # rearrange " b c h w -> b (h w) c"
+        batch_size, num_channels, height, width = shape_list(hidden_state)
+        hidden_size = height * width
+        hidden_state = tf.reshape(hidden_state, shape=(batch_size, num_channels, hidden_size))
+        hidden_state = tf.transpose(hidden_state, perm=(0, 2, 1))
+        return hidden_state
+
+
+class TFCvtSelfAttentionProjection(tf.keras.layers.Layer):
+    """Convolutional Projection for Attention."""
+
+    def __init__(
+        self,
+        config: CvtConfig,
+        kernel_size: int,
+        stride: int,
+        padding: int,
+        projection_method: str = "dw_bn",
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        if projection_method == "dw_bn":
+            self.convolution_projection = TFCvtSelfAttentionConvProjection(
+                config, kernel_size, stride, padding, name="convolution_projection"
+            )
+        self.linear_projection = TFCvtSelfAttentionLinearProjection()
+
+    def call(self, hidden_state: tf.Tensor) -> tf.Tensor:
+        # When running on CPU, `tf.keras.layers.Conv2D` doesn't support `NCHW` format.
+        # So change the input format from `NCHW` to `NHWC`.
+        hidden_state = tf.transpose(hidden_state, perm=(0, 3, 2, 1))
+        hidden_state = self.convolution_projection(hidden_state)
+        hidden_state = tf.transpose(hidden_state, perm=(0, 3, 2, 1))
+        hidden_state = self.linear_projection(hidden_state)
+        return hidden_state
+
+
+class TFCvtSelfAttention(tf.keras.layers.Layer):
+    """
+    Self-attention layer. A depth-wise separable convolution operation (Convolutional Projection), is applied for
+    query, key, and value embeddings.
+    """
+
+    def __init__(
+        self,
+        config: CvtConfig,
+        num_heads: int,
+        embed_dim: int,
+        kernel_size: int,
+        stride_q: int,
+        stride_kv: int,
+        padding_q: int,
+        padding_kv: int,
+        qkv_projection_method: str,
+        attention_drop_rate: float,
+        with_cls_token: bool = True,
+        **kwargs
+    ):
```

Comment:
The `qkv_bias` argument is missing here (and in other classes), and then in the initialization of the Dense layers